common/flatpak-dir.c

```diff
@@ -3566,6 +3576,9 @@ flatpak_dir_do_resolve_p2p_refs (FlatpakDir             *self,
   if (results == NULL)
     return FALSE;
 
+  if (results[0] == NULL)
+    return flatpak_fail (error, _("No remotes found which provide these refs: [%s]"), refs_str->str);
```

Comment:
I wonder if it would be useful with a better error code than FAILED for these.

Reply:
Fixed in https://github.com/flatpak/flatpak/pull/2895